hello_sdl_android/src/main/java/com/sdl/hellosdlandroid/SdlService.java

```diff
@@ -180,100 +96,121 @@ public void enterForeground() {
+				if (intent != null && intent.hasExtra(UsbManager.EXTRA_ACCESSORY)) { //If we want to support USB transport
+					if (android.os.Build.VERSION.SDK_INT <= android.os.Build.VERSION_CODES.HONEYCOMB) {
+						Log.e(TAG, "Unable to start proxy. Android OS version is too low");
+						return;
+					} else {
+						//We have a usb transport
+						transport = new USBTransportConfig(getBaseContext(), (UsbAccessory) intent.getParcelableExtra(UsbManager.EXTRA_ACCESSORY));
+						Log.d(TAG, "USB created.");
+					}
 				}
 			}
-		}else if(forceConnect){
-			proxy.forceOnConnected();
-		}
-	}
 
-	private void disposeSyncProxy() {
-		LockScreenActivity.updateLockScreenStatus(LockScreenStatus.OFF);
+			// The app type to be used
+			Vector<AppHMIType> appType = new Vector<>();
+			appType.add(AppHMIType.MEDIA);
+
+			// The manager listener helps you know when certain events that pertain to the SDL Manager happen
+			// Here we will listen for ON_HMI_STATUS and ON_COMMAND notifications
+			SdlManagerListener listener = new SdlManagerListener() {
+				@Override
+				public void onStart() {
+					// HMI Status Listener
+					sdlManager.addOnRPCNotificationListener(FunctionID.ON_HMI_STATUS, new OnRPCNotificationListener() {
+						@Override
+						public void onNotified(RPCNotification notification) {
+							OnHMIStatus status = (OnHMIStatus) notification;
+							if (status.getHmiLevel() == HMILevel.HMI_FULL) {
+								if (!firstShow) {
+									sendCommands();
+									performWelcomeSpeak();
+									performWelcomeShow();
+									firstShow = true;
+								}
+							}
+						}
+					});
+
+					// Menu Selected Listener
+					sdlManager.addOnRPCNotificationListener(FunctionID.ON_COMMAND, new OnRPCNotificationListener() {
+						@Override
+						public void onNotified(RPCNotification notification) {
+							OnCommand command = (OnCommand) notification;
+							Integer id = command.getCmdID();
+							if(id != null){
+								switch(id){
+									case TEST_COMMAND_ID:
+										showTest();
+										break;
+								}
+							}
+						}
+					});
+				}
 
-		if (proxy != null) {
-			try {
-				proxy.dispose();
-			} catch (Exception e) {
-				e.printStackTrace();
-			} finally {
-				proxy = null;
-			}
-		}
-		this.firstNonHmiNone = true;
-		this.isVehicleDataSubscribed = false;
-		
-	}
+				@Override
+				public void onDestroy() {
+					SdlService.this.stopSelf();
+				}
 
-	/**
-	 * Will show a sample test message on screen as well as speak a sample test message
-	 */
-	private void showTest(){
-		try {
-			proxy.show(TEST_COMMAND_NAME, "Command has been selected", TextAlignment.CENTERED, CorrelationIdGenerator.generateId());
-			proxy.speak(TEST_COMMAND_NAME, CorrelationIdGenerator.generateId());
-		} catch (SdlException e) {
-			e.printStackTrace();
+				@Override
+				public void onError(String info, Exception e) {
+				}
+			};
+
+			// Create App Icon, this is set in the SdlManager builder
+			SdlArtwork appIcon = new SdlArtwork();
```

Comment:
We can use the constructor to use even less code:
`SdlArtwork appIcon = new SdlArtwork(ICON_FILENAME, FileType.GRAPHIC_PNG, R.mipmap.ic_launcher, true);`